src/c_api/c_api.cc

```diff
@@ -174,48 +174,68 @@ int MXNDArrayCreateNone(NDArrayHandle *out) {
   API_END();
 }
 
+template<typename DataType, typename dimtype>
+void CreateNDArrayImpl(const DataType* shape,
+                       dimtype ndim,
+                       int dev_type,
+                       int dev_id,
+                       int delay_alloc,
+                       int dtype,
+                       NDArrayHandle* out) {
+  *out = new NDArray(mxnet::TShape(shape, shape + ndim),
+                     Context::Create(static_cast<Context::DeviceType>(dev_type), dev_id),
+                     delay_alloc != 0, dtype);
+}
+
 int MXNDArrayCreate(const mx_uint *shape,
                     mx_uint ndim,
                     int dev_type,
                     int dev_id,
                     int delay_alloc,
                     NDArrayHandle *out) {
   API_BEGIN();
-  *out = new NDArray(
-      mxnet::TShape(shape, shape + ndim),
-      Context::Create(static_cast<Context::DeviceType>(dev_type), dev_id),
-      delay_alloc != 0);
+  *out = new NDArray(mxnet::TShape(shape, shape + ndim),
```

Comment:
Why don't we use CreateNDArrayImpl for consistency here as well? Also it's more like a wrapper than an Impl. Maybe the "Impl" part of the name could be refined?

Reply:
@access2rohit Please also use CreateNDArrayImpl as @larroy suggested.